lib/ansible/template/__init__.py

```diff
@@ -159,6 +163,39 @@ def _escape_backslashes(data, jinja_env):
     return data
 
 
+def is_template(data, jinja_env):
+    """This function attempts to quickly detect whether a value is a jinja2
+    template. To do so, we look for the first 2 matching jinja2 tokens for
+    start and end delimiters.
+    """
+    found = None
+    start = True
+    comment = False
+    d2 = jinja_env.preprocess(data)
+
+    # This wraps a lot of code, but this is due to lex returing a generator
+    # so we may get an exception at any part of the loop
+    try:
+        for token in jinja_env.lex(d2):
+            if token[1] in JINJA2_BEGIN_TOKENS:
+                if start and token[1] == 'comment_begin':
+                    # Comments can wrap other token types
+                    comment = True
+                start = False
+                # Example: variable_end -> variable
+                found = token[1].split('_')[0]
+            elif token[1] in JINJA2_END_TOKENS:
+                if token[1].split('_')[0] == found:
+                    return True
+                elif comment:
+                    continue
+                return False
```

Comment:
won't we get false positives unless we always consume the generator?  ``{{ this is valid }} but then  {{ this is not'`

Reply:
This is based on the recommendation @jimi-c made.  It's a best effort guess at it being a template.  As long as the first token has a matching closing token as the next token, then we make the assumption it is a template.  It may fail to template later though.

Reply:
we'll get syntax error on templating later anyways